cmake/Warnings.cmake

```diff
@@ -0,0 +1,161 @@
+        #/we4289 # nonstandard extension used: 'variable': loop control variable declared in the for-loop is used outside
+        #        # the for-loop scope
+        #/w14296 # 'operator': expression is always 'boolean_value'
+        #/w14311 # 'variable': pointer truncation from 'type1' to 'type2'
+        #/w14545 # expression before comma evaluates to a function which is missing an argument list
+        #/w14546 # function call before comma missing argument list
+        #/w14547 # 'operator': operator before comma has no effect; expected operator with side-effect
+        #/w14549 # 'operator': operator before comma has no effect; did you intend 'operator'?
+        #/w14555 # expression has no effect; expected expression with side- effect
+        #/w14619 # pragma warning: there is no warning number 'number'
+        #/w14640 # Enable warning on thread un-safe static member initialization
+        #/w14826 # Conversion from 'type1' to 'type_2' is sign-extended. This may cause unexpected runtime behavior.
+        #/w14905 # wide string literal cast to 'LPSTR'
+        #/w14906 # string literal cast to 'LPWSTR'
+        #/w14928 # illegal copy-initialization; more than one user-defined conversion has been implicitly applied
+        #/permissive- # standards conformance mode for MSVC compiler.
+    )
+
+    set(CLANG_WARNINGS
+        -Wall
+        -Wextra # reasonable and standard
+        -Wshadow # warn the user if a variable declaration shadows one from a parent context
+        -pedantic
+        #-Wnon-virtual-dtor # warn the user if a class with virtual functions has a non-virtual destructor. This helps
+        #                    # catch hard to track down memory errors
+        #-Wold-style-cast # warn for c-style casts
+        #-Wcast-align # warn for potential performance problem casts
+        #-Wunused # warn on anything being unused
+        #-Woverloaded-virtual # warn if you overload (not override) a virtual function
+        #-Wpedantic # warn if non-standard C++ is used
+        #-Wconversion # warn on type conversions that may lose data
+        #-Wsign-conversion # warn on sign conversions
+        #-Wnull-dereference # warn if a null dereference is detected
+        #-Wdouble-promotion # warn if float is implicit promoted to double
+        #-Wformat=2 # warn on security issues around functions that format output (ie printf)
+    )
+
+    set(GCC_WARNINGS
+        ${CLANG_WARNINGS}
+        -Wmisleading-indentation # warn if indentation implies blocks where blocks do not exist
+        -Wduplicated-cond # warn if if / else chain has duplicated conditions
+        -Wduplicated-branches # warn if if / else branches have duplicated code
+        -Wlogical-op # warn about logical operations being used where bitwise were probably wanted
+        -Wuseless-cast # warn if you perform a cast to the same type
+    )
+
+    if(${target_set_warnings_AS_ERROR})
```

Comment:
This isn't used anywhere. Treating warnings as errors should be default behavior. 
Also, the naming is inconsistent: either use all caps or no caps.

Reply:
This same variable is used in the  next set_warnings() function and it's not used yet I guess. Needs discussion.